block-storage/src/main/scala/coop/rchain/blockstorage/BlockDagFileStorage.scala

```diff
@@ -1034,6 +1082,121 @@ object BlockDagFileStorage {
             }
     } yield LMDBStore[F](env, dbi)
 
+  private def readBlockHashesByDeploy[F[_]: Sync](
+      randomAccessIO: RandomAccessIO[F]
+  ): F[List[(DeployId, BlockHash)]] = {
+    def readRec(
+        result: List[(DeployId, BlockHash)]
+    ): F[List[(DeployId, BlockHash)]] =
+      for {
```

Comment:
It looks like the outer for can be implemented in terms of traverse

Reply:
I don't think traverse is applicable here but I simplified (i.e. got rid of "for") it by using flatMap